beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/aws/S3Client.java

```diff
@@ -73,10 +74,21 @@ List<String> deleteObjects(String bucket, List<String> keys) {
       return Collections.emptyList();
     }
     if (!dryRunEnabled) {
-      keys.forEach(key -> log.info("Deleting: \"{}/{}\"", bucket, key));
-      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket)
-          .withKeys(keys.toArray(new String[] {}));
-      DeleteObjectsResult deleteObjectsResult = amazonS3.deleteObjects(deleteObjectsRequest);
+      log.info("Attempting to delete a total of {} objects, from [{}] to [{}]", keys.size(), keys.get(0), keys.get(keys.size()-1));
+      DeleteObjectsResult deleteObjectsResult = new DeleteObjectsResult(new ArrayList<>());
+      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket);
+      int totalKeys = keys.size();
+      int chunkSize = 1000;
+      int indexStart;
+      int indexEnd = 0;
+      while(indexEnd < totalKeys) {
+        indexStart = indexEnd;
+        indexEnd = indexStart + chunkSize < totalKeys ? indexStart + chunkSize : totalKeys;
```

Comment:
It'd be better to give `indexStart + chunkSize` a variable name that has a meaning. Right now, I don't know why the sum of those needs to be smaller than `totalKeys`.
Also, you could consider adding this into a private function that houses the logic.

Reply:
moved the logic to another function and added some names to make it more readable